src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -1905,6 +1907,97 @@ withNextSrv c userId usedSrvs initUsed action = do
     writeTVar usedSrvs $! used'
   action srvAuth
 
+data AgentServersSummary = AgentServersSummary
+  { usersServersSummary :: Map UserId ServersSummary -- UserId to be mapped to User in chat core
+    -- compute in backend to not repeat logic in UI?
+    -- requires separate type / changing summary type to accomodate many servers (e.g. lists of workers, etc.)
+    -- totalServersSummary :: ServersSummary
+  }
+  deriving (Show)
+
+data ServersSummary = ServersSummary
+  { smpServersSummary :: [SMPServerSummary],
+    xftpServersSummary :: [XFTPServerSummary]
+  }
+  deriving (Show)
+
+data SMPServerSummary = SMPServerSummary
+  { smpServer :: SMPServer,
+    usedForNewConnections :: Bool,
+    subscriptionsSummary :: Maybe SMPServerSubsSummary,
+    workersSummary :: SMPServerWorkersSummary,
+    commandsStats :: [CommandStat],
+    rcvMsgCounts :: [SMPServerRcvMsgCounts],
+    deliveryInfo :: Maybe SMPServerDeliveryInfo -- Nothing if server is only used for receiving?
+  }
+  deriving (Show)
+
+data SMPServerSubsSummary = SMPServerSubsSummary
+  { activeSubscriptions :: [SMPServerSubInfo],
+    pendingSubscriptions :: [SMPServerSubInfo],
+    removedSubscriptions :: [SMPServerSubInfo]
+  }
+  deriving (Show)
+
+data SMPServerSubInfo = SMPServerSubInfo
+  { rcvId :: Text,
+    subError :: Maybe String
```

Comment:
this is basically copied from SubInfo type which is used across all three subscription lists, without userId and server. Since we're optimizing this type for "server info" view without removed subs, will remove it from this type